cmd/store.go

```diff
@@ -109,6 +110,12 @@ func (s *store) DeleteNamespace(namespace string) {
 func (s *store) checkRoleForNamespace(role string, namespace string) bool {
 	ar := s.rolesByNamespace[namespace]
 	for _, r := range ar {
+		re := regexp.MustCompile(r)
+		if re.MatchString(role) {
+			log.Debugf("Role:%s matching regexp %s on namespace:%s found.", role, r, namespace)
+			return true
+		}
+
 		if r == role {
```

Comment:
We can do away with this if statement here because a regex that is exactly the same as a role will match that role. As an aside, you need to change the README to ensure users know that the namespace role restriction annotation is interpreted as a regex since some characters in some ARNs may need escaping (e.g. *).